eng/scripts/Attest-CPEX-KPIs.ps1

```diff
@@ -0,0 +1,314 @@
+function AddAttestationEntry($targetId, $actionItemId, $status, $targetType, $url, $productName) {
+    $command = @"
+.append $TableName <|
+print
+    ActionItemId = "$actionItemId",
+    TargetId = "$targetId",
+    TargetType = "$targetType",
+    Status = int($status),
+    CreatedTime = datetime(now),
+    EvidenceUrl = "$url"
+"@
+
+    Write-Host "Adding attestation entry for product [$targetId], with status [$status] for KPI action id [$actionItemId]."
+    InvokeKustoCommand $command
+    Write-Host "Added attestation entry for product [$targetId], with status [$status] for KPI action id [$actionItemId]."
+
+    if (-not $successfulAttestations.ContainsKey($actionItemId)) {
+        $successfulAttestations[$actionItemId] = @()
+    }
+
+    $successfulAttestations[$actionItemId] += @{
+        productId = $targetId
+        productName = $productName
+        status = if ($status -eq $COMPLETED) { "Completed" } else { "Not Applicable"}
+    }
+}
+
+function SendEmailNotification($emailTo, $emailCC, $emailSubject, $emailBody) {
+    try {
+        $emailerUri = $env:azure_sdk_emailer_uri
+        $body = @{ EmailTo = $emailTo; EmailCC = $emailCC; Subject = $emailSubject; Body = $emailBody} | ConvertTo-Json -Depth 3
+        $response = Invoke-RestMethod -Uri $emailerUri -Method Post -Body $body -ContentType "application/json"
+    } catch {
+        Write-Error "Failed to send email."
+        Write-Error "To: $emailTo"
+        Write-Error "CC: $emailCC"
+        Write-Error "Subject: $emailSubject"
+        Write-Error "Body: $emailBody"
+        Write-Error "Exception message: $($_.Exception.Message)"
+    }
+}
+
+function BuildSuccessEmailBody {
+    if ($successfulAttestations.Count -eq 0) {
+        return "No products were marked as Completed or Not Applicable for any KPI in this latest run."
+    }
+
+    $body = "<h2>Successful CPEX KPI Attestations</h2>"
+    foreach ($kpiId in $successfulAttestations.Keys) {
+        if ($successfulAttestations[$kpiId].Count -ne 0) {
+            $title = $KPI_ID_TO_TITLE[$kpiId]
+            $body += "<h3> $title ($kpiId)</h3><ul>"
+            foreach ($entry in $successfulAttestations[$kpiId]) {
+                $productName = $entry.productName
+                $productId = $entry.productId
+                $status = $entry.status
+                $link = "$SERVICE_TREE_URL$productId"
+                $body += "<li><strong>$productName</strong></li> (<a href=$link>$productId</a>): $status"
+            }
+        }
+
+        $body += "</ul>"
+    }
+    
+    return $body
+}
+
+function BuildFailureEmailBody {
+    if ($failedAttestations.Count -eq 0) {
+        return "No errors occurred during the latest CPEX KPI attestation automated run. All entries were processed sucessfully."
+    }
+
+    $body = "The following errors occurred during the latest CPEX KPI attestation automated run: `n`n"
+    
+    foreach ($entry in $failedAttestations) {
+        $productName = $entry.productName
```

Comment:
Can you please include what's the type for each failed attestation? Like data plane private preview or onboarding etc.